ICD11 --> ORDO mappings

joeflack4 opened the pull request:
## About
ICD11<->ORDO Mappings

See: [Strategy](https://github.com/monarch-initiative/icd11/pull/11#discussion_r1542176048)

Here's the [google sheet for Mondo->ICD11](https://docs.google.com/spreadsheets/d/1npe1PdZrb0YTtAbPqgu7OZ961g7KwC0_eZLAlK7RCEc/edit#gid=1894001781).

## Changes
ORDO Mappings
- Add: New release output: ordo-icd11foundation.sssom.tsv
  - Completed work in mappings.py
  - Add: utils.py for this: This file now has several functions that are exactly the same in several repositories. Needs to be put into a package.
- Update: config/icd11.sssom-metadata.yml: removed some unnecessary entries. Added necessary ones.
- Add: requirements*.txt files: for installation of Python packages; and updated readme about that.
- Delete: Temporary anlalysis files in src/compare_*/
- Add: icd11mms-exact-matches.sparql for querying ordo.owl for mappings. But this may not be necessary and may get removed.
- Add: src/temp_compare_matches_owl_and_nomenclature.py: Temporary file for analysis for comparing Ordo->ICD11MMS exact matches from nomenclature file and from OWL release.
- Add: New release output: mondo-icd11foundation.sssom.tsv

## Related
- #11

---

CC @souzadevinicius This is a PR to get ICD11Foundation<->Orphanet mappings, from Orphanet. We don't have standalone Orphanet repo, so I added them here. We can use these to get Mondo->ICD11Foundation mappings.

joeflack4 committed:
Temp: ORDO mappings analysis
Exploratory analysis to try to find the best way to get ORDO<->ICD11Foundation mappings.

joeflack4 committed:
Temp: ORDO mappings analysis
Exploratory analysis to try to find the best way to get ORDO<->ICD11Foundation mappings.

joeflack4 committed:
ORDO Mappings
- Add: New release output: ordo-icd11.sssom.tsv
  - Completed work in mappings.py
  - Add: utils.py for this: This file now has several functions that are exactly the same in several repositories. Needs to be put into a package.
- Update: config/icd11.sssom-metadata.yml: removed some unnecessary entries. Added necessary ones.
- Add: requirements*.txt files: for installation of Python packages; and updated readme about that.
- Delete: Temporary anlalysis files in src/compare_*/
- Add: icd11mms-exact-matches.sparql for querying ordo.owl for mappings. But this may not be necessary and may get removed.
- Add: src/temp_compare_matches_owl_and_nomenclature.py: Temporary file for analysis for comparing Ordo->ICD11MMS exact matches from nomenclature file and from OWL release.
- Bug fix: run.sh: Updated to an older version that works.

joeflack4 commented on makefile:
```diff
@@ -35,6 +36,19 @@ release: | tmp/output/release/
+xxx: tmp/input/Orphanet_Nomenclature_Pack_EN/ORPHA_ICD11_mapping_en_newversion_2023.xml
+
+# Mappings
+tmp/input/Orphanet_Nomenclature_Pack_EN/ORPHA_ICD11_mapping_en_newversion_2023.xml: | tmp/input/
```
## Issue: Prerequisite dependency will break in the future (possible fix)
I want to make this reproducible. For that, the "API" I'm interacting with needs to be stable, but in this case there is a file source issues.

### Unstable things
**File name**
I need to use `ORPHA_ICD11_mapping_en_newversion_2023.xml`. However, there are two things that are unstable about this filename: (i) `_2023`, and (ii) `_newversion`. I imagine that in the future (i) will change according to the year / date, and (ii) will probably be dropped from the filename or renamed. I think this 'new version' may supercede the old version of the file eventually.

**Download URL**
Further, I don't have a URL for this release specifically; it doesn't exist yet. I only have the URL for the 'latest' release: https://www.orphadata.com/data/nomenclature/packs/Orphanet_Nomenclature_Pack_EN.zip. Whenever 2023 does get archived as a previous release, it should be accessible via this URL: https://www.orphadata.com/data/nomenclature/previous_versions/Orphanet_Nomenclature_Pack_EN_2023.zi

### Possible fix
See [comment below](https://github.com/monarch-initiative/icd11/pull/12#discussion_r1543742223).

matentzn replied:
Thank you for this analysis.. This is a bit of a pity, but in our case, we can use your pipeline for the 1 time alignment, and then work with Orphanet to provide a stable mapping.

joeflack4 replied:
@matentzn Actually, it may be the case that it will be stable soon. Julie just sent me an email, and she said that the ICD11Foundation references will also appear in the alignment file, which has a stable URL and filename, starting July this year. I've set myself a reminder on my calendar to check.

matentzn commented on makefile:
```diff
@@ -26,6 +26,7 @@ tmp/input/:
 
 tmp/input/source.owl: tmp/input/source.gz
 	gunzip -c $< > $@
+	rm $<
```
Probably nothing but `rm` in conjunction with a variable always makes me scared

matentzn commented on makefile:
```diff
@@ -35,6 +36,20 @@ release: | tmp/output/release/
+	wget http://www.orphadata.org/data/ORDO/ordo_orphanet.owl -O $@
+
+tmp/output/release/ordo-icd11.sssom.tsv: tmp/input/Orphanet_Nomenclature_Pack_EN/ORPHA_ICD11_mapping_en_newversion_2023.xml tmp/input/ordo.owl | tmp/output/release/
+	python3 src/make_sssom.py $< > $@
```
wrong python filename?

joeflack4 replied:
Ah yes, I was still in the very early stages yesterday. It's updated now.

matentzn reviewed (commented):
Before you proceed with this:

what mappings are in ORDO that are not in the "nomenclature pack"?

joeflack4 commented on src/icd11mms-exact-matches.sparql:
```diff
@@ -0,0 +1,22 @@
+    ECO:0000218 ?mapping_precision_string .
+
+  FILTER(STRSTARTS(STR(?mapping_precision_string), "- E (Exact mapping: the two concepts are equivalent)."))
+  FILTER(STRSTARTS(STR(?xref), "ICD-11:"))
```
## FYI: `icd11mms-exact-matches.sparql`
I tried using OAK at first, but from what I can see, `SqlImplementation` does not support querying axioms for 'mapping triples' (e.g. `oboInOwl:hasDbXref`), only 'relationship triples'  (e.g. `rdfs:subClassOf`).

joeflack4 replied:
This actually ended up not being necessary. I did not need to use it to create the SSSOM output. I only needed the nomenclature file, not the OWL. But I also used this in an analysis to compare mappings between those two sources.

joeflack4 replied:
I think ample time has been given to look at this, and a determination was made that this file is not necessary, so gonna mark this resolved.

joeflack4 commented on src/mappings.py:
```diff
@@ -0,0 +1,157 @@
+
+    # Get only columns we care about
+    df = df[['subject_id', 'subject_label', 'predicate_id', 'object_id']]
+    write_sssom(df, input_sssom_config, outpath)
```
## Output
- [x] Review @twhetzel @matentzn 

### Results
[ordo-icd11foundation.sssom](https://docs.google.com/spreadsheets/d/1KZAeAyqYtjk7y-p7H7RUmBTT5rWX0KlBTdqlbX03Gc8/edit#gid=621796084)

I've also uploaded it to the [latest release](https://github.com/monarch-initiative/icd11/releases/tag/v2024-03-30).

Example:
|subject_id|subject_label|predicate_id|object_id|
|-|-|-|-|
|Orphanet:58|Alexander disease|skos:exactMatch|icd11.foundation:2023359698|
|Orphanet:213|Cystinosis|skos:exactMatch|icd11.foundation:733715856|
|Orphanet:447|Paroxysmal nocturnal hemoglobinuria|skos:exactMatch|icd11.foundation:859588467|

<details><summary>SSSOM mapping set metadata</summary>
<p>

```
# creator_id:   0000-0002-2906-7319
# curie_map:
#     Orphanet: http://www.orpha.net/ORDO/Orphanet_
#     icd11.foundation: http://id.who.int/icd/entity/
#     skos: http://www.w3.org/2004/02/skos/core#
# license:   http://w3id.org/sssom/license/unspecified
```

</p>
</details> 

### Discussion
I ended up not needing the OWL release to get exact mappings. The nomenclature XML had them. However, as with [difference of set of terms between (a) ORDO OWL and (b)nomenclature file(s)](https://github.com/monarch-initiative/icd11/pull/11#discussion_r1543773843), there is also a [difference in exact matches between ORDO OWL and nomenclature file](https://github.com/monarch-initiative/icd11/pull/12#discussion_r1545495600). I'm not worried about these discrepancies because these two sources are released on different schedules (6 and 12 months).

matentzn replied:
- [x] please make sure the mapping set is valid sssom using `sssom validate`, see

https://github.com/mapping-commons/mh_mapping_initiative/blob/e6154b97ef04becaf707ff89b9a79817c835ac4e/Makefile#L22

as an example.

I am assuming object_label is not available!

matentzn replied:
- [x] also add `mapping_provider` property with a link to the orphanet website)

joeflack4 replied:
Done and done!
@matentzn While working on the `sssom validate` bit, I noticed a problem, so I made an issue:
- https://github.com/mapping-commons/sssom-py/issues/512

@twhetzel FYI

joeflack4 replied:
Can't believe I forgot to do this before, but here's this output as a google sheet: [ordo-icd11foundation.sssom](https://docs.google.com/spreadsheets/d/1KZAeAyqYtjk7y-p7H7RUmBTT5rWX0KlBTdqlbX03Gc8/edit#gid=621796084)

joeflack4 commented on src/temp_compare_matches_owl_and_nomenclature.py:
```diff
@@ -0,0 +1,30 @@
+owl_set = set_tups(owl_df)  # n=1,459
+i = nom_set.intersection(owl_set)  # n=1,323
+n_diff = nom_set.difference(owl_set)  # n=17
+o_diff = owl_set.difference(nom_set)  # n=136
```
## Analysis: Difference in exact matches between ORDO OWL and nomenclature file
- [x] FYI / review @matentzn @twhetzel 

### Results
I ended up not needing to query the Orphanet OWL release to get exact matches and then do lookups in the nomenclature XML. I was able to query the XML directly. I did an analysis though to compare the exact matches between the two different sources, and they are as follows.

Note that this comparison is for ICD11MMS exact matches, as the foundation references do not exist (yet) in the OWL release, but ICD11MMS refs exist in both. I was originally going to use the ICD11MMS refs as an intermediary proxy to get Orphanet->foundation refs.

I'm not too worried about the discrepancy between these sources, as the OWL is released on a 6 months schedule and the nomenclature files are on 12 months, so that is what I'm attributing the difference to.

- n exact matches in nomenclature file(s): 1,340
- n exact matches in OWL release: 1,459
- n in intersection: 1,323
- n nomenclature files(s) not in OWL: 17
- n in OWL not in nomenclature file(s): 136

---

See also: [Analysis: Difference of ORDO OWL to nomenclature file(s)](https://github.com/monarch-initiative/icd11/pull/11#discussion_r1543773843)

### Action item: Filtering?
- [x] @twhetzel @matentzn Decide if I should do this

If you want, since the OWL is released more often, I could potentially utilize it. It's possible that when the OWL gets released after a nomenclature release, some exact matches will be dropped. It looks like right now the number of cases for that is 17 (_n nomenclature files(s) not in OWL: 17_). So, if incorporate querying of the OWL into this pipeline, then some dropped exact matches (in this case, 17 of them) will sneak in. Of course, this will just be rectified 6 months later when the new nomenclature file comes out, so I don't know how big of a deal this is.

### Finally
- [x] @joeflack4 When this has been reviewed, if no further steps are needed, and if Orphanet OWL querying continues to not be needed, remove those files / code from the codebase.

matentzn replied:
Fantastic analysis! Thank you very much. I think the difference is small enough to resort to OWL, which means easier maintenance moving forward! (my opinion, feel free to contradict)

joeflack4 replied:
Glad this was helpful. You mis-wrote something though or possibly misunderstood something. It sounds like you're asking me to change things so that I use the OWL instead of the XML for these matches. I do agree that if that could be done, that'd be easier maintenance. But it can't be done. The OWL does not have any ICD11Foundation references. Its only potential role here would be to help us do some additional filtering on what's returned back from the XML.

In any case, given that the XML must be utilized, and given that you agree that this is a small / insignificant difference, it looks like the determination is to opt out of that additional filtering, and keep what I've done as is. So that's what I'm going with!

joeflack4 commented on makefile:
```diff
@@ -6,7 +6,7 @@ SOURCE_URL=https://icd11files.blob.core.windows.net/tmp/whofic-2023-04-08.owl.gz
 
 # MAIN COMMANDS / GOALS ------------------------------------------------------------------------------------------------
-all: tmp/output/release/icd11foundation.owl
+all: tmp/output/release/icd11foundation.owl tmp/output/release/ordo-icd11.sssom.tsv
```
## GitHub action for release?
- [ ] Make a determination @matentzn or @twhetzel 

Now there are two artefacts in [the release](https://github.com/monarch-initiative/icd11/releases/tag/v2024-03-30):
1. `icd11foundation.owl`: This is about the same as the source we download, but with unicode characters stripped.
2. `ordo-icd11.sssom.tsv`

Prior to this PR, the reason to not create a GitHub action is because the URL tied to `icd11foundation.owl` is to a static, past release, as opposed to a 'latest' release that gets updated regularly. We do want to change this if we can though; see:
- #8

In the case of `ordo-icd11.sssom.tsv`, though, this will be updatable every July, as that is when the nomenclature files are released. So we could theoretically have an action that runs and detects a new nomenclature file and then makes a release. We could set this to run every year in July/August. Or, since the exact release date is variable, and if we want to get the release out quickly after they do their release, we could set it this action to run like daily/weekly, and if it detects a new nomenclature release, it does its thing, else quits. So it would run often but only make a new release once a year.

joeflack4 replied:
Going to check for updated nomenclature manually the next time it is released. Will make an issue to make a GH action.

joeflack4 replied:
Created an issue for this:
- #13

joeflack4 commented on src/utils.py:
```diff
@@ -0,0 +1,56 @@
+    os.remove(temp_filtered_config_path)
+
+
+def write_tsv_with_comments(df: pd.DataFrame, comments_file: Union[Path, str], outpath: Union[Path, str]):
```
## General: `write_tsv_with_comments()`
@matentzn Let me know what the existing best practice is supposed to be for this.
I have this function (and several others it's used with) in the repos for GARD, [MedGen](https://github.com/monarch-initiative/medgen/pull/18#discussion_r1536735796), ICD11, a TIMS repo, and possibly elsewhere. Maybe there's a function in `sssom-py` that does this like I want? I have a dataframe and a path to a `sssom-metadata.yml`, and I want to write that to a file and include the comment at the top.

joeflack4 replied:
- [x] @joeflack4 Nico advised in another PR what to do here ([1](https://github.com/monarch-initiative/medgen/pull/18#discussion_r1536735796), [2](https://github.com/monarch-initiative/medgen/pull/18#discussion_r1536736041), [3](https://github.com/monarch-initiative/medgen/pull/18#discussion_r1551603841)); what standard `sssom-py` practices to employ, so I will need to do that here.

joeflack4 replied:
Done (also made the same changes in my [Medgen PR](https://github.com/monarch-initiative/medgen/pull/18#discussion_r1554796042)). And in doing so, I also encountered some possible bugs and UX issues, and made:
- https://github.com/mapping-commons/sssom-py/issues/513
- https://github.com/mapping-commons/sssom-py/issues/514
- https://github.com/mapping-commons/sssom-py/issues/515
- https://github.com/mapping-commons/sssom-py/issues/516
- https://github.com/mapping-commons/sssom-py/issues/517

joeflack4 replied:
This should be ready to be marked resolved by a reviewer.

Nico has examined these 5 issues as well.

joeflack4 commented on src/temp_compare_matches_owl_and_nomenclature.py:
```diff
@@ -0,0 +1,30 @@
+"""Temporary analysis file to compare exact matches to ICD11MMS between Orphanet OWL release and nomenclature files."""
```
## Complete review for: [Temp: ORDO mappings analysis #11](https://github.com/monarch-initiative/icd11/pull/11#issue-2209486198)
- [ ] @matentzn and/or @twhetzel review

I've closed that PR because it does not need to be merged. It serves as simply a way to communicate an analysis. As of 2024/03/30, there are still 5 open discussion threads there, so I'd like to make sure those are at least looked at.

joeflack4 committed:
ORDO Mappings
- Delete: Unnecessary files and goals: temp analysis, getting of ordo.owl and querying it via SPARQL.
- Add: SSSOM validation
- Update: Replaced ad hoc SSSOM utilities with the ones from sssom-py
- Add: mondo-icd11 mappings
- Rename: SSSOM files: icd11 -> icd11foundation

matentzn commented on makefile:
```diff
@@ -1,12 +1,19 @@
+mappings: mappings-validate
+
+mappings-validate: tmp/output/release/ordo-icd11.sssom.tsv
+	sssom validate $<
```
greeat!

matentzn commented on a.txt:
wrong commit

joeflack4 replied:
Thanks for catching that. Removed

matentzn reviewed (approved):
LGTM!

matentzn reviewed (commented):
Change my review to comment due to the wrongly commented a.txt file

joeflack4 commented on src/mappings.py:
```diff
@@ -0,0 +1,168 @@
+        metadata: Dict = yaml.load(yaml_file, Loader=yaml.FullLoader)
+    converter = curies.Converter.from_prefix_map(metadata['curie_map'])
+    msdf: MappingSetDataFrame = MappingSetDataFrame(converter=converter, df=df, metadata=metadata)
+    with open(outpath, 'w') as f:
```
## `ordo_exactmatch_icd11foundation.sssom.tsv`; what to do with it?
Here's the: [google sheet](https://docs.google.com/spreadsheets/d/1KZAeAyqYtjk7y-p7H7RUmBTT5rWX0KlBTdqlbX03Gc8/edit#gid=621796084). I'm releasing this file in the ICD11 repo, but I don't think I've written anything in `mondo-ingest` or `mondo` to grab it yet. What should we do with it? Should nothing be done with? Was it always meant to be an intermediary to generate [`mondo-icd11foundation.sssom.tsv`](https://github.com/monarch-initiative/icd11/pull/12#discussion_r1566558267)? If so, I need to remove it from the release.

related: [mondo-icd11foundation.sssom.tsv; what to do with it?](https://github.com/monarch-initiative/icd11/pull/12#discussion_r1566562502)

twhetzel replied:
@matentzn ^^

joeflack4 replied:
These were only needed as an intermediary to create Mondo mappings. They're nice to have though. Leave them.

joeflack4 commented on config/icd11.sssom-metadata.yml:
```diff
@@ -0,0 +1,15 @@
+  skos: http://www.w3.org/2004/02/skos/core#
+  sssom: https://w3id.org/sssom/
+license: http://w3id.org/sssom/license/unspecified
+mapping_provider: https://www.orpha.net/
```
## `mapping_provider` for **Mondo -(ORDO)--> ICD11**
- [ ] @matentzn advise

We're doing `mapping_provider: https://www.orpha.net/` for the ORDO->ICD11 mappings, but the MONDO->ICD11 mappings require inputs from Orphanet as well as `mondo.sssom.tsv`. I've left the `mapping_provider` the same, but I have a feeling it should be changed to something else.

joeflack4 replied:
Indeed probably a composite `mapping_provider` would make sense, but not to worry.

joeflack4 commented on makefile:
```diff
@@ -35,6 +43,36 @@ release: | tmp/output/release/
+	wget https://www.orphadata.com/data/nomenclature/packs/Orphanet_Nomenclature_Pack_EN.zip -O tmp/input/Orphanet_Nomenclature_Pack_EN.zip
+	unzip tmp/input/Orphanet_Nomenclature_Pack_EN.zip -d tmp/input/Orphanet_Nomenclature_Pack_EN
+
+tmp/output/release/mondo-icd11.sssom.tsv tmp/output/release/ordo-icd11.sssom.tsv: tmp/input/Orphanet_Nomenclature_Pack_EN/ORPHA_ICD11_mapping_en_newversion_2023.xml tmp/input/mondo.sssom.tsv | tmp/output/release/
```
## `mondo_exactmatch_icd11foundation.sssom.tsv`: Review
- [x] @sabrinatoro Sanity check review

---

Here's the: [google sheet](https://docs.google.com/spreadsheets/d/1npe1PdZrb0YTtAbPqgu7OZ961g7KwC0_eZLAlK7RCEc/edit#gid=1894001781)

On [Slack](https://monarchinitiative.slack.com/archives/C03AW548J8M/p1713181767278319), @matentzn asked for:
> Get a headstart on the mapping by doing ICD11->ORDO->Mondo (i.e. merge in ICD11->Mondo asap from that)
> we need the sssom file in a google sheet with label so that someone can do a sanity check. Then we just merge it.

I created this by first getting the ORDO->ICD11 mappings, then doing a join on exact mappings from `mondo.sssom.tsv`.

Related:
- [`ordo-icd11foundation.sssom.tsv`; what to do with it?](https://github.com/monarch-initiative/icd11/pull/12#discussion_r1566376036)
- [`mondo-icd11foundation.sssom.tsv`; what to do with it?](https://github.com/monarch-initiative/icd11/pull/12#discussion_r1566562502)

joeflack4 replied:
This has been reviewed, and is now utilized post mondo-ingest build.

joeflack4 commented on makefile:
```diff
@@ -35,6 +43,36 @@ release: | tmp/output/release/
+	wget https://www.orphadata.com/data/nomenclature/packs/Orphanet_Nomenclature_Pack_EN.zip -O tmp/input/Orphanet_Nomenclature_Pack_EN.zip
+	unzip tmp/input/Orphanet_Nomenclature_Pack_EN.zip -d tmp/input/Orphanet_Nomenclature_Pack_EN
+
+tmp/output/release/mondo-icd11.sssom.tsv tmp/output/release/ordo-icd11.sssom.tsv: tmp/input/Orphanet_Nomenclature_Pack_EN/ORPHA_ICD11_mapping_en_newversion_2023.xml tmp/input/mondo.sssom.tsv | tmp/output/release/
```
## `mondo_exactmatch_icd11foundation.sssom.tsv`; what to do with it?
Here's the: [google sheet](https://docs.google.com/spreadsheets/d/1npe1PdZrb0YTtAbPqgu7OZ961g7KwC0_eZLAlK7RCEc/edit#gid=1894001781). I've also added it to a new release.

Related: [`mondo-icd11foundation.sssom.tsv`: Review](https://github.com/monarch-initiative/icd11/pull/12#discussion_r1566558267)

FYI @twhetzel 

### Questions
- [ ] 1. **@matentzn Should a robot template for this come next / instead?**

I'm imagining something similar to the one in this MedGen PR: https://github.com/monarch-initiative/medgen/pull/19 ([google sheet](https://docs.google.com/spreadsheets/d/1OD7WUR-nJ9lUqRv01TTghmf3aGvQRbWV5ZhVhthzB6U/edit#gid=1627855620)). If we do want to create a robot template for this, do we still need the SSSOM file?

- [x] 2. **@matentzn What to do with it?**

You asked for a SSSOM file, but I don't know what to do with it now that it is created. I don't know if it should somehow go into `mondo-ingest` or `mondo`.

If `mondo`, put it in `src/ontology/mappings/`?
```
cd src/ontology/mappings; ls
mondo.sssom.tsv                       mondo_exactmatch_icd10who.sssom.tsv   mondo_hasdbxref_icd10cm.sssom.tsv     mondo_hasdbxref_obi.sssom.tsv         mondo_narrowmatch_orphanet.sssom.tsv
```
If, so, I think I need to rename it to `mondo_exactmatch_icd11foundation.sssom.tsv`, yes?

Also, if I do this, then this would be the only SSSOM file in the `mondo` repo that is copied over into this directory from an external repository. All of the other files there are generated by some process I'm currently a little in the dark on.

twhetzel replied:
I'll need to leave this to Nico since I don't know how this is handled yet.

joeflack4 replied:
Sabrina did ask us for a robot template. For now, she has created it on her own. There is still the question as to whether or not I should automate that.

joeflack4 replied:
Nothing is done with the SSSOM file itself. It's just an intermediary format for the robot template.

- [ ] Consider: general tool for converting SSSOM to robot template
  - Perhaps in `mondolib`. Could be a `make` command.
  - Sometimes there will be additional fields that need to get added to the robot template. Would not be part of this conversion. Would be added post facto.

joeflack4 commented on makefile:
```diff
@@ -35,6 +43,37 @@ release: | tmp/output/release/
+	cd tmp/input/ &&\
+	git clone --depth 1 https://github.com/monarch-initiative/mondo
+
+tmp/input/mondo.sssom.tsv: tmp/input/mondo/
```
## Goal for `mondo.sssom.tsv`
### FYI: `make` goal differences from `mondo-ingest` approach
I deviated a bit from what was in `mondo-ingest`. What I have here makes more sense to me (in general & for this specific use case) compared to how `mondo-ingest` does it:
```make
USE_MONDO_RELEASE=false

tmp/mondo_repo_built:
	if [ $(USE_MONDO_RELEASE) = true ]; then wget http://purl.obolibrary.org/obo/mondo.owl -O $@; else cd $(TMPDIR) &&\
		rm -rf ./mondo/ &&\
		git clone --depth 1 https://github.com/monarch-initiative/mondo &&\
		cd mondo/src/ontology &&\
		make mondo.owl mappings -B MIR=false IMP=false MIR=false &&\
		cd ../../../../ &&\
		touch $@; fi

$(TMPDIR)/mondo.sssom.tsv: $(TMPDIR)/mondo_repo_built
	cp $(TMPDIR)/mondo/src/ontology/mappings/mondo.sssom.tsv $@

```

### `USE_PREBUILT=true`
I wanted to get the latest `mondo.sssom.tsv` from the `mondo` repo by cloning the repo and running `make mondo.owl mappings`, but I got an error:
`cp: cannot stat 'tmp/input/mondo/src/ontology/mappings/mondo.sssom.tsv': No such file or directory`

Not sure why the error; I checked and the file at that path does exist.

For now, I just set this to `true`, and it simply downloads the `mondo.sssom.tsv` from the latest commit in `mondo`.

joeflack4 commented on src/mappings.py:
```diff
@@ -0,0 +1,213 @@
+        .merge(mondo_sssom_df, how='inner', on='ordo_id')[[
+            'subject_id', 'subject_label', 'predicate_id', 'object_id', 'object_label', 'ordo_id',
+            'mapping_justification']]
+        .rename(columns={'ordo_id': 'match_string'}))
```
## `match_string`
- [x] @matentzn review

@matentzn asked for this [on Slack](https://monarchinitiative.slack.com/archives/C03AW548J8M/p1713300259182229?thread_ts=1713225589.907679&cid=C03AW548J8M) and I completed and reported there. But I wasn't sure if I did it correctly.

As asked, I added the ORDO IDs that were the intermediary for the Mondo<->ICD mappings.

I wrote:
> Maybe it’s good as-is. But if you want it formalized, I guess I could add a `subject_match_field` with values `skos:exactMatch` or something.
> I’m looking at this file as a reference for `match_string`, but this is how it’s used in lexmatch, so I guess that’s different from our case here

- [Link to `match_string` docs](https://mapping-commons.github.io/sssom/match_string/)


<details><summary>Example use of match_string in mondo-ingest</summary>
<p>


|subject_id|subject_label|object_id|predicate_id|object_label|mapping_justification|mapping_tool|confidence|subject_match_field|object_match_field|match_string|comment|predicate_modifier|reviewer_id|rejection_reason|
|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|
|MONDO:0005571|polycythemia|NCIT:C27794|skos:exactmatch|Polycythemia (Excluding Polycythemia Vera)|semapv:LexicalMatching|oaklib|0.8497788952|rdfs:label|rdfs:label|polycythemia|LEXMATCH|Not|https://orcid.org/0000-0002-4142-7153|mapping rule incorrect|
|MONDO:0016642|meningioma|DOID:0080842|skos:exactmatch|intracranial meningioma|semapv:LexicalMatching|oaklib|0.8|oio:hasExactSynonym|rdfs:label|intracranial meningioma|LEXMATCH|Not|https://orcid.org/0000-0002-4142-7153|synonym type incorrect|
|MONDO:0016715|ependymoblastoma|DOID:0081286|skos:exactmatch|embryonal tumor with multilayered rosettes|semapv:LexicalMatching|oaklib|0.8|oio:hasExactSynonym|rdfs:label|embryonal tumor with multilayered rosettes|LEXMATCH|Not|https://orcid.org/0000-0002-4142-7153|split/merge difference in Mondo vs source|

</p>
</details>

joeflack4 replied:
The better way in SSSOM would be to model how we do "chained mappings", e.g. a->b and b->c, therefore a->c.

So since that hasn't been formalized yet, we're using this `match_string` field in an unconventional way to represent that.
